src/OrleansRuntime/Placement/IPlacementContext.cs

```diff
@@ -54,22 +56,32 @@ internal interface IPlacementContext
         /// <returns></returns>
         bool TryGetActivationData(ActivationId id, out ActivationData activationData);
 
-        void GetGrainTypeInfo(int typeCode, out string grainClass, out PlacementStrategy placement, string genericArguments = null);
+        void GetGrainTypeInfo(int typeCode, out string grainClass, out PlacementStrategy placement, out MultiClusterRegistrationStrategy strategy, string genericArguments = null);
     }
 
     internal static class PlacementContextExtensions
     {
-        public static Task<List<ActivationAddress>> Lookup(this IPlacementContext @this, GrainId grainId)
+        public static async Task<List<ActivationAddress>> Lookup(this IPlacementContext @this, GrainId grainId)
         {
             List<ActivationAddress> l;
-            return @this.FastLookup(grainId, out l) ? Task.FromResult(l) : @this.FullLookup(grainId);
+            if (@this.FastLookup(grainId, out l))
+            {
+                return l;
+            }
+            else
+            {
+                var result = await @this.FullLookup(grainId);
```

Comment:
Ideally, we would avoid allocating a `Task` (for the fast path) as well as the async state machine. It helps if `FullLookup` returns a result which is directly consumable by the caller